### DISCUSSION TITLE
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'

### DISCUSSION THREAD

**Discussion by janiceylau** [talmolab/sleap#1259] [Help!]
Hi! I added more videos to a project and wanted to train a new model. However, I came across the following error which did not occur with my previous videos. What could be the reason? I found the same error message in this post: [https://github.com/talmolab/sleap/discussions/961](url). Does it mean one/some video file(s) is/are corrupted? But I can see every frame from the newly added videos at the SLEAP GUI.

![Screenshot from 2023-04-04 18-59-52](https://user-images.githubusercontent.com/102581787/229864727-a13215de-3749-4f6d-8e41-43f61fd0e394.png)

That block of error messages then repeated three times...

Thanks for your help!

**Comment by janiceylau** [comment]
Just saw this error at the terminal which is for the GUI (instead of the error above which is from running the python scripts). Though it happened some time ago and when I opened the GUI again with that project, the same error did not reproduce.

![Screenshot from 2023-04-04 19-25-11](https://user-images.githubusercontent.com/102581787/229872858-6fb442fc-0d9b-47f0-b9d5-d060ff9f14ee.png)

**Comment by rmorgan7125** [comment]
Hi,
I ran across this issue when I added a new video accidentally and moved it to a different directory.
I could open sleap-label, but it would give me a prompt saying it could not find the video and ask me to update it. 
When I updated the new directory though, it would never save apparently and I would get this same error when starting to train, so I was forced to use a video in the current directory, which is very strange.
Sorry if this is not the most applicable to your issue, but this is how I resolved it on my end.

**Reply by janiceylau** [reply]
Hi! What do you mean by "forced to use a video in the current directory"? Did you solve the problem by moving the video (back?) to the current directory (where the label is?) instead of using it in the new directory?

**Comment by roomrys** [comment]
In another case #1339, the user did not unzip the directory containing the videos, hence SLEAP was unable to load the videos correctly.

**Comment by janiceylau** [comment]
Hi I came across this problem again. I labelled the videos at my local computer (since that is faster), then I copied the label file to the remote server, and needed to update the new paths to the videos at the server. The label, the video folder and the sleap folder are now under the same directory. But I still have this problem...

**Reply by talmo** [reply]
Do you mind taking a screenshot of the folder so we can see exactly how it's organized?

Also maybe you could paste the result of running this command (replacing `labels.slp` with the filename of your labels):

```
python -c "import sleap; print(sleap.load_file('labels.slp').videos)"
```

Thanks!

Talmo

**Reply by janiceylau** [reply]
Hi Talmo, 

Thanks for your reply! Here is the screenshot of the folder: 
![Screenshot from 2023-08-06 16-59-57](https://github.com/talmolab/sleap/assets/102581787/8013728c-1656-47e3-ba48-7e459623c97f)
where the folders sleap, videos_cows_top (with 4 videos), and the file labels....slp are involved. 

And the screenshot after running the command: 
![Screenshot from 2023-08-06 16-53-37](https://github.com/talmolab/sleap/assets/102581787/4b7a50db-cad1-46a2-b9a8-a3090e0005f4)

Thanks very much!

**Reply by roomrys** [reply]
Hi @janiceylau,

Is the path listed for the videos in your screenshot the same as the path where your videos now reside (i.e. the directory you have open in the screenshot above)? I would expect these paths to be different if you are having problems loading frames from the video...

Thanks,
Liezl

**Comment by roomrys** [comment]
Hi @janiceylau,

You mentioned in your first two posts that one error was from running python scripts and the second post was from running the GUI. I am wondering if the video paths are only being updated if you 
1. open the GUI, 
2.  find them, and then 
3. save the project. 

I believe we have added a bit of code since 1.3.0 (definitely since 1.3.0a0) for saving video paths when opening a project via the CLI. Can you verify whether you run into this problem if you follow the steps above (which I would expect to work regardless of version)?

Thanks,
Liezl

**Reply by janiceylau** [reply]
Hi Liezl, 

Thanks very much for your reply. Yes I updated the video paths through the steps you mentioned. The version shown at the GUI is 1.3.0a0. Right I trained the model by running the GUI and the scripts (for finetuning) respectively. Sometimes the error occurred at both places but sometimes only at the GUI. I guess I will simply try out some other label files and see if the same problem reproduces, since I also modified some lines of the codes it may be hard to say where the problems really lie.

**Reply by roomrys** [reply]
Argh the PRs I was thinking of are #1098 and #1148 which are both included in 1.3.0a0.... Thanks for keeping the thread updated.
